examples/ensemble/plot_feature_transformation.py

```diff
@@ -70,24 +70,15 @@
 y_pred_rf_lm = rf_lm.predict_proba(rf_enc.transform(rf.apply(X_test)))[:, 1]
 fpr_rf_lm, tpr_rf_lm, _ = roc_curve(y_test, y_pred_rf_lm)
 
-
-# Supervised transformation based on gradient boosted trees. Demonstrates
-# the use of each tree's apply() method.
-def gradient_apply(clf, X):
-    X_trans = []
-    for tree in clf.estimators_.ravel():
-        X_trans.append(tree.apply(X))
-    return np.array(X_trans).T
-
 grd = GradientBoostingClassifier(n_estimators=n_estimator)
 grd_enc = OneHotEncoder()
 grd_lm = LogisticRegression()
 grd.fit(X_train, y_train)
-grd_enc.fit(gradient_apply(grd, X_train))
-grd_lm.fit(grd_enc.transform(gradient_apply(grd, X_train_lr)), y_train_lr)
+grd_enc.fit(grd.apply(X_train)[:, :, 0])
+grd_lm.fit(grd_enc.transform(grd.apply(X_train_lr)[:, :, 0]), y_train_lr)
 
 y_pred_grd_lm = grd_lm.predict_proba(
-    grd_enc.transform(gradient_apply(grd, X_test)))[:, 1]
+    grd_enc.transform(grd.apply(X_test)[:, :, 0]))[:, 1]
 fpr_grd_lm, tpr_grd_lm, _ = roc_curve(y_test, y_pred_grd_lm)
 
 
```

Comment:
I think it would be great to do a second plot that zooms on the top left corner of the first plot to see the difference among the best models:

```python
plt.xlim(0, 0.2)
plt.ylim(0.8, 1)
```